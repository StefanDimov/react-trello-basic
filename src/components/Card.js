import React from 'react'

export default class Card extends React.Component {
    render() {
        const { title } = this.props
        return (
            <div>{title}</div>
        )
    }
}

const { string } = React.PropTypes
Card.propTypes = {
    title: string.isRequired
}
