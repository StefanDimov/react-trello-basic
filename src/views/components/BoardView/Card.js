import React from 'react'
import * as cardUtils from '../../../utils/cardUtils'
import { Glyphicon } from 'react-bootstrap'

/**
 * Component for visualizing a card title with some icons depending on the data it holds.
 * Ment to be used in CardList Component
 * @class comp/Card
 */
export default class Card extends React.Component {
    render() { // eslint-disable-line
        const { title, description } = this.props.card
        return (
            <div>
                <span>{title}</span>

                <span className="pull-right">
                    { description && <Glyphicon glyph="align-left" /> }
                </span>
            </div>
        )
    }
}

Card.propTypes = {
    card: cardUtils.cardPropType.isRequired
}
