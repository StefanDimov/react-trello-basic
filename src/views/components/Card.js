import React from 'react'
import cardUtils from '../../utils/cardUtils'
import { Glyphicon } from 'react-bootstrap'

export default class Card extends React.Component {
    render() {
        const { title, description } = this.props.card
        return (
            <div>
                <span>{title}</span>

                <span className="pull-right">
                    { description && <Glyphicon glyph="align-left" /> }
                </span>
            </div>
        )
    }
}

Card.propTypes = {
    card: cardUtils.cardPropType.isRequired
}
