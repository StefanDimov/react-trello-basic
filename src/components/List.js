import React from 'react'
import Card from './Card'
import { Panel, ListGroup, ListGroupItem } from 'react-bootstrap'

export default class List extends React.Component {
    render() {
        const {title, cards, onCardClick, children} = this.props

        const cardElements = cards.map(card =>
            <ListGroupItem onClick={onCardClick} key={card.title}>
                <Card title={card.title}/>
            </ListGroupItem>
        )

        return (
            <Panel header={title}>
                {(cardElements.length > 0) && <ListGroup>{cardElements}</ListGroup>}
                {children}
            </Panel>
        )
    }
}

const { string, array, func } = React.PropTypes
List.propTypes = {
    title: string.isRequired,
    cards: array.isRequired,
    onCardClick: func.isRequired
}
