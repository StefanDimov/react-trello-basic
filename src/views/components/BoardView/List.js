import React from 'react'
import Card from './Card'
import { Panel, ListGroup, ListGroupItem } from 'react-bootstrap'

import listUtils from '../../../utils/listUtils'

export default class List extends React.Component {
    render() {
        const { list, onCardClick, children} = this.props
        const { title, cards } = list

        const cardElements = cards.map(card =>
            <ListGroupItem onClick={() => onCardClick(card)} key={card.id}>
                <Card card={card}/>
            </ListGroupItem>
        )

        return (
            <Panel header={title} className="my-list-container-item">
                {(cardElements.length > 0) && <ListGroup>{cardElements}</ListGroup>}
                {children}
            </Panel>
        )
    }
}

List.propTypes = {
    list: listUtils.listPropType.isRequired,
    onCardClick: React.PropTypes.func.isRequired
}
