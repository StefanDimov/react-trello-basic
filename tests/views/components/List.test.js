import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import List from '../../../src/views/components/List'
import Card from '../../../src/views/components/Card'
import { Panel, ListGroup, ListGroupItem } from 'react-bootstrap'

import { getCardOnlyWithIds } from '../../_mocks/Card.mocks'

describe('List', () => {

    let list
    let onCardClick = jest.fn()
    const cardWithId = getCardOnlyWithIds(1)
    const anotherCardWithId = getCardOnlyWithIds(2)

    beforeEach(() => {
        list = { id: 'uniqueId', cards: [] }
        onCardClick.mockClear()
    })

    describe('rendering', () => {

        it('should render correctly with title', () => {
            list.title = 'title'
            const tree = renderer.create(
                <List list={list} onCardClick={onCardClick} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('should render correctly with children', () => {
            const tree = renderer.create(
                <List list={list} onCardClick={onCardClick}>
                    <span>unique</span>
                </List>
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('should render correctly with a card', () => {
            list.cards = [cardWithId]
            const tree = renderer.create(
                <List list={list} onCardClick={onCardClick} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('should render correctly with two card', () => {
            list.cards = [cardWithId, anotherCardWithId]
            const tree = renderer.create(
                <List list={list} onCardClick={onCardClick} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('should render correctly with no cards', () => {
            list.cards = []
            const tree = renderer.create(
                <List list={list} onCardClick={onCardClick} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })
    })

    describe('cards', () => {

        it('should each be passed to the onCardClick func on ListGroupItem click', () => {
            list.cards = [cardWithId, anotherCardWithId]
            const wrapper = shallow(<List list={list} onCardClick={onCardClick} />)

            wrapper.find(ListGroupItem).first().simulate('click')
            expect(onCardClick).toHaveBeenLastCalledWith(cardWithId)

            wrapper.find(ListGroupItem).last().simulate('click')
            expect(onCardClick).toHaveBeenLastCalledWith(anotherCardWithId)
        })
    })
})
