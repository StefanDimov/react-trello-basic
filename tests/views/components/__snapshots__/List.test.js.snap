exports[`List rendering should render correctly with a card 1`] = `
<div
  className="my-list-container-item panel panel-default"
  id={undefined}>
  <div
    className="panel-body">
    <div
      className="list-group">
      <button
        className="list-group-item"
        onClick={[Function]}
        type="button">
        <div>
          <span />
          <span
            className="pull-right" />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`List rendering should render correctly with children 1`] = `
<div
  className="my-list-container-item panel panel-default"
  id={undefined}>
  <div
    className="panel-body">
    <span>
      unique
    </span>
  </div>
</div>
`;

exports[`List rendering should render correctly with no cards 1`] = `
<div
  className="my-list-container-item panel panel-default"
  id={undefined} />
`;

exports[`List rendering should render correctly with title 1`] = `
<div
  className="my-list-container-item panel panel-default"
  id={undefined}>
  <div
    className="panel-heading">
    title
  </div>
</div>
`;

exports[`List rendering should render correctly with two card 1`] = `
<div
  className="my-list-container-item panel panel-default"
  id={undefined}>
  <div
    className="panel-body">
    <div
      className="list-group">
      <button
        className="list-group-item"
        onClick={[Function]}
        type="button">
        <div>
          <span />
          <span
            className="pull-right" />
        </div>
      </button>
      <button
        className="list-group-item"
        onClick={[Function]}
        type="button">
        <div>
          <span />
          <span
            className="pull-right" />
        </div>
      </button>
    </div>
  </div>
</div>
`;
